mlfinlab/ensemble/sb_bagging.py

```diff
@@ -0,0 +1,660 @@
+            t[1] for t in all_results))
+
+        if self.oob_score:
+            self._set_oob_score(X, y)
+
+        return self
+
+    def _get_estimators_indices(self):
+
+        # Get indicator matrix
+        subsampled_ind_mat = self.ind_mat[:, self.timestamp_int_index_mapping.loc[self.X_time_index]]
+
+        # Get drawn indices along both sample and feature axes
+        for seed in self._seeds:
+            # Operations accessing random_state must be performed identically
+            # to those in `_parallel_build_estimators()`
+            random_state = np.random.RandomState(seed)
+            feature_indices, sample_indices = _generate_bagging_indices(
+                random_state, self.bootstrap_features,
+                self.n_features_, self._max_features, self._max_samples, subsampled_ind_mat)
+
+            yield feature_indices, sample_indices
+
+    def estimators_samples_(self):
+        """The subset of drawn samples for each base estimator.
+        Returns a dynamically generated list of indices identifying
+        the samples used for fitting each member of the ensemble, i.e.,
+        the in-bag samples.
+        Note: the list is re-created at each call to the property in order
+        to reduce the object memory footprint by not storing the sampling
+        data. Thus fetching the property may be slower than expected.
+        """
+        return [sample_indices
+                for _, sample_indices in self._get_estimators_indices()]
+
+
+class SequentiallyBootstrappedBaggingClassifier(SequentiallyBootstrappedBaseBagging, BaggingClassifier,
+                                                ClassifierMixin):
+    """A Sequentially Bootstrapped Bagging classifier.
+    A Sequentially Bootstrapped Bagging classifier is an ensemble meta-estimator that fits base
+    classifiers each on random subsets of the original dataset generated using
+    Sequential Bootstrapping sampling procedure and then aggregate their individual predictions (
+    either by voting or by averaging) to form a final prediction. Such a meta-estimator can typically be used as
+    a way to reduce the variance of a black-box estimator (e.g., a decision
+    tree), by introducing randomization into its construction procedure and
+    then making an ensemble out of it.
+    This algorithm encompasses several works from the literature. When random
+    subsets of the dataset are drawn as random subsets of the samples, then
+    this algorithm is known as Pasting [1]_. If samples are drawn with
+    replacement, then the method is known as Bagging [2]_. When random subsets
+    of the dataset are drawn as random subsets of the features, then the method
+    is known as Random Subspaces [3]_. Finally, when base estimators are built
+    on subsets of both samples and features, then the method is known as
+    Random Patches [4]_.
+    Read more in the :ref:`User Guide <bagging>`.
+    Parameters
+    ----------
```

Comment:
The format of the argument description in the docstring here is different from other files in this PR, and in the module. Is there a reason for that? Will sphinx pick this up ok?